text/0000-index-assign.md

```diff
@@ -0,0 +1,325 @@
+- Feature Name: indexed_assignment
+- Start Date: 2015-05-19
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add a `IndexAssign` trait that allows overloading "indexed assignment" expressions like `a[b] = c`.
+
+# Motivation
+
+Let users define syntactic sugar for operations like these:
+
+- Inserting a *new* key-value pair into a map
+
+``` rust
+let mut map = HashMap::new();
+map[key] = value;  // equivalent to `map.insert(key, value);`
+```
+
+- Setting each element of a "slice" to some value
+
+(slice, as in a fraction of a collection, which may not necessarily be stored in contiguous memory)
+
+``` rust
+let mut matrix = { .. };
+
+// Set each element of the second row of `matrix` to `1`
+matrix[1] = 1;
+
+// or
+
+let mut vector = { .. };
+
+// set first 4 elements of `vector` to zero
+vector[..4] = 0;
+```
+
+- Copying a slice into another
+
+``` rust
+// Copy the third row of `another_matrix` into the second row of `matrix`
+matrix[1] = &another_matrix[2]
```

Comment:
Aha I see! I'll just write it down so that I can think about it: :)

`matrix[1]` = &another_matrix[2]`

 `IndexAssign::index_assign(self: &mut Row([T]), index: usize, rhs: &Row([T]))`